addons/mail/models/mail_activity.py

```diff
@@ -39,6 +40,12 @@ class MailActivityType(models.Model):
         help='Categories may trigger specific behavior like opening calendar view')
 
 
+    @api.multi
+    def unlink(self):
+        if self.env['mail.activity'].search_count([('activity_type_id', 'in', self.ids)]):
+            raise UserError(_("This activity type is already used in scheduled activites. If you no longer want to use it, you can archive it."))
+        return super(MailActivityType, self).unlink()
+
```

Comment:
@tde-banana-odoo Here we are re-implementing an `ondelete='restrict'` for the sake of change the error message... not sure it is worthy

Reply:
Seems right.